### ISSUE TITLE
buildifier: warn on glob (non-)metacharacters other than *

### ISSUE THREAD

**Issue by alandonovan** [bazelbuild/buildtools#604]
Within Google's code base we've noticed a number of uses of glob metacharacters such as \ [ ] ? . that assume POSIX glob semantics, whereas Bazel's glob has only one metacharacter, *, and all other characters are treated literally. We've even seen some users assuming regexp semantics (e.g. `\.`) It's tempting to disallow all these non-metachars completely, occasionally these symbols do appear within file names.

Buildifier should warn on any use of these (non-)metacharacters in glob. Bonus marks if it suppresses the check when these literally match an actual file name.

**Comment by laurentlb** [comment]
There's a proposed change to globs (b/19665064 inside Google; I'll move it to GitHub).
The idea is to make the glob function fail when any of its patterns matches nothing, because it's often due to a typo in the pattern. We also add a pattern to revert to the old behavior, in case it was actually intended.

Would that solve your problem?

**Comment by alandonovan** [comment]
Yes, that's a good solution. (I assume you mean "add a parameter" not "add a pattern".)
But I suspect there are quite a lot of globs in the wild (in macros) that expect no matches, and would need the new parameter to be set.
